src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,187 @@
+            throw new IllegalArgumentException("상행역과 하행역 모두 노선에 포함되어 있으므로 추가할 수 없습니다.");
+        }
+    }
+
+    private void validateNotContainsStation(Section section) {
+        if (!existsStation(section.getUpStation()) && !existsStation(section.getDownStation())) {
+            throw new IllegalArgumentException("상행역과 하행역이 모두 노선에 포함되지 않으므로 추가할 수 없습니다.");
+        }
+    }
+
+    private boolean isUpTerminus(Section section) {
+        return section.getDownStation().equals(getUpStation()) && !existsStation(section.getUpStation());
+    }
+
+    private boolean isDownTerminus(Section section) {
+        return section.getUpStation().equals(getDownStation()) && !existsStation(section.getDownStation());
+    }
+
+    private void decideForkedLoad(Section section) {
+        if (isFrontForkedLoad(section)) {
+            changeFrontSection(section);
+            return;
+        }
+
+        if (isBackForkedLoad(section)) {
+            changeBackSection(section);
+            return;
+        }
+
+        throw new IllegalArgumentException("새로운 구간의 길이가 기존 구간의 길이 보다 크거나 같으므로 추가할 수 없습니다.");
+    }
+
+    private boolean isFrontForkedLoad(Section section) {
+        if (existsStation(section.getUpStation())) {
+            Section frontSection = findByUpStation(section.getUpStation());
+            return frontSection.getDistance() > section.getDistance();
+        }
+
+        return false;
+    }
+
+    private void changeFrontSection(Section section) {
+        Section frontSection = findByUpStation(section.getUpStation());
+        frontSection.changeUpStation(section);
+        int frontIndex = value.indexOf(frontSection);
+        value.remove(frontSection);
+        value.add(frontIndex, section);
+        value.add(frontIndex + 1, frontSection);
+    }
+
+    private boolean isBackForkedLoad(Section section) {
+        if (existsStation(section.getDownStation())) {
+            Section backSection = findByDownStation(section.getDownStation());
+            return backSection.getDistance() > section.getDistance();
+        }
+
+        return false;
+    }
+
+    private void changeBackSection(Section section) {
+        Section backSection = findByDownStation(section.getDownStation());
+        backSection.changeDownStation(section);
+        int backIndex = value.indexOf(backSection);
+        value.add(backIndex + 1, section);
+    }
+
+    private boolean existsStation(Station station) {
+        return value.stream()
+                .anyMatch(section -> section.existsStation(station));
+    }
+
+    public void remove(Station station) {
+        if (value.size() == 1) {
+            throw new IllegalArgumentException("구간이 1개만 존재하므로 삭제가 불가능 합니다.");
+        }
+
+        if (getUpStation().equals(station)) {
+            value.remove(value.get(0));
+            return;
+        }
+
+        if (getDownStation().equals(station)) {
+            value.remove(value.get(value.size() - 1));
+            return;
+        }
```

Comment:
꺼내서(get) 비교하지마시고, 객체에게 메시지를 보내보세요

```suggestion
        if (isUpstation(station)) {
            value.remove(value.get(0));
            return;
        }

        if (isDownStation(station)) {
            value.remove(value.get(value.size() - 1));
            return;
        }
```